core/virtualhostname/virtualhostname.go

```diff
@@ -74,6 +130,28 @@ func (i Info) Target() HostnameTarget {
 	return i.target
 }
 
+// String returns the virtual hostname stringified.
+func (i Info) String() string {
+	switch i.target {
+	case MachineTarget:
+		return fmt.Sprintf("%d.%s.%s", i.machine, i.modelUUID, Domain)
+	case UnitTarget:
+		parts := strings.Split(i.unit, "/")
+		if len(parts) != 2 {
+			panic("invalid unit name") // this shouldn't happen because we have validated the unit in the constructor.
+		}
+		return fmt.Sprintf("%s.%s.%s.%s", parts[1], parts[0], i.modelUUID, Domain)
```

Comment:
Can you name parts[1] and parts[0] above, so that it's easier to read. Same for container targets.
```
unitNumber := parts[1]
unitName := parts[0]
```